integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -19,17 +19,89 @@
 import { expect } from "chai";
 import Realm from "realm";
 
-import { openRealmBefore } from "../hooks";
+import { openRealmBeforeEach } from "../hooks";
+import { sleep } from "../utils/sleep";
 
 type Item<ValueType = Realm.Mixed> = {
   dict: Realm.Dictionary<ValueType>;
 };
+const DictWrongSchema: Realm.ObjectSchema = {
+  name: "Dictionary",
+  properties: {
+    a: "wwwww{}",
+  },
+};
+const DictSchema: Realm.ObjectSchema = {
+  name: "Dictionary",
+  properties: {
+    fields: "{}",
+  },
+};
+const Child: Realm.ObjectSchema = {
+  name: "Children",
+  properties: {
+    num: "int",
+  },
+};
+const TwoDictSchema: Realm.ObjectSchema = {
+  name: "Dictionary",
+  properties: {
+    dict1: "{}",
+    dict2: "{}",
+  },
+};
+const EmbeddedChild = {
+  name: "Children",
+  embedded: true,
+  properties: {
+    num: "int",
+  },
+};
+
+const DictTypedSchema = {
+  name: "TypedDictionary",
+  properties: {
+    dict1: { type: "dictionary", objectType: "Children" }, // dictionary of objects is nullable by default
+    dict2: { type: "dictionary", objectType: "Children", optional: true },
+  },
+};
+
+const DictMixedSchema = {
+  name: "MixedDictionary",
+  properties: {
+    dict1: "{}",
+    dict2: "{}",
+  },
+};
 
+type IDictSchema = {
+  fields: Record<any, any>;
+};
+type ITwoDictSchema = {
+  dict1: Record<any, any>;
+  dict2: Record<any, any>;
+};
+type IEmbeddedChild = {
+  num: number;
+};
+type IDictTypedSchema = {
+  dict1: Record<string, IEmbeddedChild>;
+  dict2: Record<string, IEmbeddedChild | null>;
+};
 type DictValues = { [key: string]: unknown };
 
 describe("Dictionary", () => {
+  describe("schema", () => {
+    it("throws on invalid schema", () => {
+      expect(() => {
+        new Realm({ schema: [DictWrongSchema] });
+      }).throws(
+        "Schema validation failed due to the following errors:\n- Property 'Dictionary.a' of type 'dictionary' has unknown object type 'wwwww'",
+      );
+    });
+  });
   describe("with unconstrained (mixed) values", () => {
-    openRealmBefore({
+    openRealmBeforeEach({
```

Comment:
I'm curious why this was needed. Every test might as well create their own `Item` object to use in their test and avoid cross talking that way.